web-admin/src/features/projects/user-management/ShareProjectPopover.svelte

```diff
@@ -0,0 +1,138 @@
+  import { Button } from "@rilldata/web-common/components/button";
+  import type { V1UserInvite } from "@rilldata/web-admin/client";
+  import UserInviteOrganization from "./UserManagementOrganization.svelte";
+  import UserInviteUserSetRole from "./UserManagementUserSetRole.svelte";
+  import {
+    Popover,
+    PopoverContent,
+    PopoverTrigger,
+  } from "@rilldata/web-common/components/popover";
+  import AvatarListItem from "../../organizations/users/AvatarListItem.svelte";
+  import UserManagementGroup from "./UserManagementGroup.svelte";
+
+  export let organization: string;
+  export let project: string;
+
+  let open = false;
+
+  $: copyLink = `${$page.url.protocol}//${$page.url.host}/${organization}/${project}`;
+
+  $: currentUser = createAdminServiceGetCurrentUser();
+  $: listProjectMemberUsergroups =
+    createAdminServiceListProjectMemberUsergroups(organization, project);
+  $: listProjectMemberUsers = createAdminServiceListProjectMemberUsers(
+    organization,
+    project,
+  );
+  $: listProjectInvites = createAdminServiceListProjectInvites(
+    organization,
+    project,
+  );
+
+  $: projectMemberUserGroupsList =
+    $listProjectMemberUsergroups.data?.members ?? [];
+  $: projectMemberUsersList = $listProjectMemberUsers.data?.members ?? [];
+  $: projectInvitesList = $listProjectInvites.data?.invites ?? [];
+
+  function coerceInvitesToUsers(invites: V1UserInvite[]) {
+    return invites.map((invite) => ({
+      ...invite,
+      userName: null,
+      userEmail: invite.email,
+      userPhotoUrl: null,
+      roleName: invite.role,
+    }));
+  }
+
+  $: usersWithPendingInvites = [
+    ...projectMemberUsersList,
+    ...coerceInvitesToUsers(projectInvitesList),
+  ];
+
+  $: showOrganizationSection =
+    projectMemberUserGroupsList.length === 1 &&
+    projectMemberUserGroupsList[0].groupName === "all-users";
+
+  $: showGroupsSection =
+    projectMemberUserGroupsList.length > 0 &&
+    projectMemberUserGroupsList.length === 1 &&
+    projectMemberUserGroupsList[0].groupName !== "all-users";
+</script>
+
+<Popover bind:open>
+  <PopoverTrigger asChild let:builder>
+    <Button builders={[builder]} type="secondary" selected={open}>Share</Button>
+  </PopoverTrigger>
+  <PopoverContent align="end" class="w-[520px] p-4">
+    <div class="flex flex-col">
+      <div class="flex flex-row items-center mb-4">
+        <div class="text-sm font-medium">{project}</div>
+        <div class="grow"></div>
+        <CopyInviteLinkButton {copyLink} />
+      </div>
+      <UserInviteForm {organization} {project} />
+      {#if showOrganizationSection}
+        <div class="mt-4">
+          <div class="text-xs text-gray-500 font-semibold uppercase">
+            Organization
+          </div>
+          <div class="flex flex-col gap-y-1">
+            <UserInviteOrganization
+              {organization}
+              {project}
+              group={projectMemberUserGroupsList[0]}
```

Comment:
Here and above, it looks like we're making the assumption that the first user group is the "all-users" group. To be safe & more explicit, could we explicitly get the "all-users" group by finding where `groupName === "all-users"`

Reply:
There is already a check above for the Organization section.

```
  $: showOrganizationSection =
    projectMemberUserGroupsList.length === 1 &&
    projectMemberUserGroupsList[0].groupName === "all-users";
```

Reply:
Yeah, so I think this makes two unsafe assumptions:
- only 1 group is returned from the API
- "all-users" is the first group in the list

What if there are >1 groups? We should still show the `OrganizationItem`, right?

I haven't inspected the API logic, but let's just say there's another group named "aaa". If the API ever returns ["aaa", "all-users"], then `showOrganizationSelection` should still be `true`.